### PR TITLE
added three new endpoints for payment method custom fields

### PR DESCRIPTION
**Pull request by holkra**

### CODE REVIEW

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentMethodResource.java [hunk 1]
```diff
@@ -270,6 +275,53 @@ public Response deletePaymentMethod(@PathParam("paymentMethodId") final String p
         return Response.status(Status.OK).build();
     }
 
+    @Timed
+    @GET
+    @Path("/{paymentMethodId:" + UUID_PATTERN + "}/" + CUSTOM_FIELDS)
+    @Produces(APPLICATION_JSON)
+    @ApiOperation(value = "Retrieve payment method custom fields", response = CustomFieldJson.class, responseContainer = "List")
+    @ApiResponses(value = {@ApiResponse(code = 400, message = "Invalid payment id supplied")})
```
Comment (pierre) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentMethodResource.java]: Typo: payment **method**

**Comment by pierre** [comment]
Can you [add tests](https://github.com/killbill/killbill/blob/master/profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestPaymentMethod.java) using your changes in https://github.com/killbill/killbill-client-java/pull/19? Don't worry about the pom.xml / parent pom.xml changes, we'll take care of it.

File: jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentMethodResource.java (hunk 1 above)
Comment (holkra) [inline jaxrs/src/main/java/org/killbill/billing/jaxrs/resources/PaymentMethodResource.java]: Fixed.

**Comment by holkra** [comment]
Ok, I will add some tests on Monday.

**Comment by holkra** [comment]
I have added a test for the new endpoint.

File: profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestPaymentMethod.java [hunk 2]
```diff
@@ -85,25 +93,73 @@ public void testPaymentMethodsPagination() throws Exception {
...
+            {
+                setObjectId(objectId);
+                setObjectType(objectType);
+                setName(name);
+                setValue(value);
+            }
+        };
+    }
+
+    private CustomField createCustomField(final UUID paymentMethodId, final String name, final String value) throws KillBillClientException {
```
Comment (pierre) [inline profiles/killbill/src/test/java/org/killbill/billing/jaxrs/TestPaymentMethod.java]: Unused (I'll fix it).

**Pull request merged**